lm75bd/lm75bd.c

```diff
@@ -25,15 +37,69 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
   return ERR_CODE_SUCCESS;
 }
 
-error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
-  
-  return ERR_CODE_SUCCESS;
+/*
+ * @brief Reads temperature from LM75BD
+ *
+ * @param devAddr - Address of the device to communicate to  
+ * @param temp - Float to store temperature in
+ *
+ */
+error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {    
+    // Check that temp has been initalized
+    if ( temp == NULL ) {
+        return ERR_CODE_UNKNOWN;  
+    }
+    uint8_t targetRegister = LM75BD_POINTER_TEMP;
+    error_code_t errCode;
+    // Select sensors internal temperature register using pointer register    
+    RETURN_IF_ERROR_CODE(i2cSendTo(devAddr, &targetRegister, sizeof(targetRegister)));
+    // XXX: p.8 of datasheet suggests first temperature read is always incorrect
+    
+    // Perform an I2C read of the device
+    uint8_t readTempBuf[2] = {0};   
+    RETURN_IF_ERROR_CODE(i2cReceiveFrom(devAddr, readTempBuf, sizeof(readTempBuf)));  
+    // Check for out-of-range error  
+    if ( (readTempBuf[1] & 0xF8) > 0 ) {  
```

Comment:
shouldn't you be checking the lower 5 bits? so `0x1F`?

Reply:
since those are the ones expected to be 0

Reply:
Good catch, twisted myself up between looking at the integration tests, my ascii art, and the datasheet :sweat_smile: